.github/workflows/cpp.yml

```diff
@@ -0,0 +1,59 @@
+name: C++ Tests
+
+on:
+  push:
+    branches: [ main, 'pr/*' ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - uses: DoozyX/clang-format-lint-action@v0.14
+      with:
+        source: '.'
+        extensions: 'h,hpp,cpp'
+        clangFormatVersion: 9
+    - uses: DoozyX/clang-format-lint-action@v0.14
+      with:
+        source: '.'
+        extensions: 'h,hpp,cpp'
+        clangFormatVersion: 14
+
+  test-cpu:
+    runs-on: ubuntu-latest
+    container: ghcr.io/wdmapp/oneapi-dpcpp-ubuntu-22.04:20230104
+    env:
+      DEBIAN_FRONTEND: noninteractive
+      LD_LIBRARY_PATH: /opt/intel/oneapi/tbb/2021.8.0/env/../lib/intel64/gcc4.8:/opt/intel/oneapi/mpi/2021.8.0//libfabric/lib:/opt/intel/oneapi/mpi/2021.8.0//lib/release:/opt/intel/oneapi/mpi/2021.8.0//lib:/opt/intel/oneapi/mkl/2023.0.0/lib/intel64:/opt/intel/oneapi/debugger/2023.0.0/gdb/intel64/lib:/opt/intel/oneapi/debugger/2023.0.0/libipt/intel64/lib:/opt/intel/oneapi/debugger/2023.0.0/dep/lib:/opt/intel/oneapi/compiler/2023.0.0/linux/lib:/opt/intel/oneapi/compiler/2023.0.0/linux/lib/x64:/opt/intel/oneapi/compiler/2023.0.0/linux/lib/oclfpga/host/linux64/lib:/opt/intel/oneapi/compiler/2023.0.0/linux/compiler/lib/intel64_lin
```

Comment:
don't you have an env to the source? Will be less noise and either to maintain

I was using that at some point https://github.com/TApplencourt/OvO/blob/master/.github/workflows/ci.yml

Reply:
I tried changing the docker image to set the appropriate env vars in entry point, and never got it to work. Agreed it would be cleaner that way, but I think better to get tests running ASAP and improved later. Will benefit gtensor as well. Using an image with oneapi already installed greatly speeds up CI running.